### PR TITLE
Merge Merkledb into master

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

Subsequent merkledb development will happen in master.

**The commits has been individually reviewed, see the corresponding PRs.**

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
d1b22a0 is probably to be reverted 📔

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@ branches:
     - /^v\d+\.\d+(\.\d+)?(-\S*)?$/
     # … and release branches, e.g. jlc/v0.2.0-release
     - /^.+-release$/
+    # temporary added for merkledb branch
+    - merkledb
```
Comment (skletsun) [inline .travis.yml]: > d1b22a0 is probably to be reverted 📔

Do we need to revert two lines above only? The rest is probably required for building on OSX.

File: exonum-java-binding/core/rust/Cargo.toml [hunk 2]
```diff
@@ -37,3 +37,10 @@ rpath = true
 
 [profile.release]
 rpath = true
+
+# FIXME: using git dependency until new Exonum with MerkleDB will be released
+[patch.crates-io]
+exonum = { git = "https://github.com/exonum/exonum", rev = "2868b8b" }
+exonum-testkit = { git = "https://github.com/exonum/exonum", rev = "2868b8b" }
+exonum-configuration = { git = "https://github.com/exonum/exonum", rev = "2868b8b" }
+exonum-time = { git = "https://github.com/exonum/exonum", rev = "2868b8b" }
```
Comment (skletsun) [inline exonum-java-binding/core/rust/Cargo.toml]: Probably it worth applying changes from #1079? It also has btc anchoring disabled but uses `exonum 0.12`.

File: .travis.yml (hunk 1 above)
Comment (dmitry-timofeev) [reply .travis.yml]: Yes, I judged on the commit titles

File: exonum-java-binding/core/rust/Cargo.toml (hunk 2 above)
Comment (dmitry-timofeev) [reply exonum-java-binding/core/rust/Cargo.toml]: It can be done in master, I think. There are questions to the present 1079.

**Comment by dmitry-timofeev** [comment]
Remaining todos:
```
$ git diff master merkledb | grep -i '+.\+\(todo\|fixme\)' 
+# FIXME: using git dependency until new Exonum with MerkleDB will be released
+//extern crate exonum_btc_anchoring; // FIXME: uncomment when anchoring uses merkledb
+//use exonum_btc_anchoring::ServiceFactory as BtcAnchoringServiceFactory; // FIXME: uncomment when anchoring uses merkledb
+        //        BTC_ANCHORING_SERVICE => Box::new(BtcAnchoringServiceFactory) as Box<ServiceFactory>, // FIXME: uncomment when anchoring uses merkledb
+    #[ignore] // FIXME: temporary ignored until anchoring service switches to MerkleDb
+    #[ignore] // FIXME: temporary ignored until anchoring service switches to MerkleDb
+      // TODO: Change message after https://jira.bf.local/browse/ECR-3354
```

**Comment by dmitry-timofeev** [comment]
Merged 🎉 

When the present PRs targeting merkledb are ready, please change the base to `master` and merge it into them.

**Pull request merged**